tests/integration-tests/conftest.py

```diff
@@ -312,6 +313,60 @@ def _cluster_factory(cluster_config, extra_args=None, raise_on_error=True):
         factory.destroy_all_clusters(test_passed=request.node.rep_call.passed)
 
 
+@pytest.fixture(scope='class')
+def api_server_factory(cfn_stacks_factory, request, public_ecr_image_uri, api_definition_s3_uri):
+    """Creates a factory for deploying API servers on-demand to each region."""
+    api_servers = {}
+
+    def _api_server_factory(server_region):
+        api_stack_name = generate_stack_name("integ-tests-api", request.config.getoption("stackname_suffix"))
+        stack_template_path = os.path.join("..", "..", "api", "infrastructure", "parallelcluster-api.yaml")
```

Comment:
can be done separately but we want to use the template passed through the --api-infrastructure-s3-uri test-runner arg so that we decouple the artifacts under test from the checked-out version of integ-tests framework. This is particularly useful when testing released artifacts.

Reply:
This parameter didn't exist in my branch, so I added it and made it an override (so that things still work easily locally).